Cgen error: genSym fails to make unique identifier for ref object types

The following program defines two ref object types with a genSym()ed id, with different named members,
and accesses the members. 

This is the nim equivalent of the generated code:

```

type
  env_4836073 = ref object
    a: int

var f: env_4836073
echo f.a

type
  env_4840010 = ref object
    b: int

var f: env_4840010
echo f.b
```

This is the actual code showing the error:


```
import macros                                         
                                                                         
macro flop(name: static string) =                                 
                                                           
  let id = genSym(nskType, "env")           
                                             
  let r =                                          
                                                           
    # type T = object                                                     
    #    a: int                                                                                                                 
                                                                                                                                
    nnkStmtList.newTree(                                                                                                        
      nnkTypeSection.newTree(                                                                                                   
        nnkTypeDef.newTree(                                         
          id,                                                          
          newEmptyNode(),                                       
          nnkRefTy.newTree(
            nnkObjectTy.newTree(                
              newEmptyNode(),                                        
              newEmptyNode(),                                            
              nnkRecList.newTree(                 
                nnkIdentDefs.newTree(                
                  newIdentNode(name),                   
                  newIdentNode("int"),
                  newEmptyNode()
                )
              )
            ) 
          )
        )
      ),

      # var f: T
 
      nnkVarSection.newTree(
        nnkIdentDefs.newTree(
          newIdentNode("f"),
          id,
          newEmptyNode()
        )
      ),

      # echo f.a

      nnkCommand.newTree(
        newIdentNode("echo"),
        nnkDotExpr.newTree(
          newIdentNode("f"),
          newIdentNode(name)
        )
      )
    )
  echo r.repr
  r


block:
  flop("a")
  
block:
  flop("b")

```

Result:


```
/home/ico/.cache/nim/t_d/@mt.nim.c:221:54: error: ‘tyObject_envcolonObjectType___f12Z9cZMf7n0MDelndFh4Yg’ {aka ‘struct tyObject_envcolonObjectType___f12Z9cZMf7n0MDelndFh4Yg’} has no member named ‘b’
  221 |   T4_[0] = nimIntToStr((*f__V2MKllaKVLp2pL5SbJiy7Q_2).b);
      |                                                      ^
Error: execution of an external compiler program 'gcc -c  -w -fmax-errors=3   -I/home/ico/external/Nim/lib -I/home/ico/sandbox/prjs/cps -o /home/ico/.cache/nim/t_d/@mt.nim.c.o /home/ico/.cache/nim/t_d/@mt.nim.c' failed with exit code: 1
```

```
Nim Compiler Version 1.3.5 [Linux: amd64]
Compiled at 2020-07-29
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: c645dba038f3d801afe1b9c842bd49bd9455b410
active boot switches: -d:release -d:danger
```

--- a/tests/macros/tmacros_various.nim
+++ b/tests/macros/tmacros_various.nim
@@ -23,6 +23,8 @@ x: some string
 ([("key", "val"), ("keyB", "2")], [("val", "key"), ("2", "keyB")])
 ([("key", "val"), ("keyB", "2")], [("val", "key"), ("2", "keyB")])
 0
+0
+0
 '''
 """
 
@@ -270,3 +272,61 @@ block: # bug #13511
 
     debugAst:
       add(foo(), component)
+
+block: # bug #15118
+  macro flop(name: static string) =
+    let id = genSym(nskType, "env")
+    let r =
+      nnkStmtList.newTree(
+        nnkTypeSection.newTree(
+          nnkTypeDef.newTree(
+            id,
+            newEmptyNode(),
+            nnkRefTy.newTree(
+              nnkObjectTy.newTree(
+                newEmptyNode(),
+                newEmptyNode(),
+                nnkRecList.newTree(
+                  nnkIdentDefs.newTree(
+                    newIdentNode(name),
+                    newIdentNode("int"),
+                    newEmptyNode()
+                  )
+                )
+              ) 
+            )
+          )
+        ),
+
+        # var f: T
+  
+        nnkVarSection.newTree(
+          nnkIdentDefs.newTree(
+            newIdentNode("f"),
+            id,
+            newEmptyNode()
+          )
+        ),
+
+        # echo f.a
+        nnkCommand.newTree(
+          newIdentNode("new"),
+          newIdentNode("f")
+        ),
+
+        nnkCommand.newTree(
+          newIdentNode("echo"),
+          nnkDotExpr.newTree(
+            newIdentNode("f"),
+            newIdentNode(name)
+          )
+        )
+      )
+    r
+
+
+  block:
+    flop("a")
+
+  block:
+    flop("b")

